src/overflow.rs

```diff
@@ -652,10 +659,15 @@ impl<'a> Context<'a> {
         );
         result.push_str(self.ident);
         result.push_str(prefix);
-        if !self.context.use_block_indent()
-            || (self.context.inside_macro() && !items_str.contains('\n') && fits_one_line)
-            || (is_extendable && extend_width <= shape.width)
-        {
+        let force_single_line = if self.context.config.version() == Version::Two {
+            !self.context.use_block_indent() || (is_extendable && extend_width <= shape.width)
+        } else {
+            let fits_one_line = items_str.len() + 2 <= shape.width;
```

Comment:
I think the comment line 643 belongs to this and should be moved along.

Reply:
Thanks! Fixed & rebased.